dali/pipeline/operator/common.h

```diff
@@ -207,8 +207,9 @@ int GetShapeArgument(TensorListShape<out_ndim> &out_tls,
                       const ArgumentWorkspace &ws,
                       int ndim = out_ndim,
                       int batch_size = -1 /* -1 = get from "batch_size" arg */) {
-  ndim = GetShapeLikeArgument<ArgumentType>(out_tls.shape, spec, argument_name, ndim, batch_size);
-  out_tls.resize(out_tls.shape().num_samples / ndim, ndim);
+  ndim = GetShapeLikeArgument<ArgumentType>(out_tls.shapes, spec, argument_name,
+                                            ws, ndim, batch_size);
+  out_tls.resize(ndim == 0 ? 0 : out_tls.shapes.size() / ndim, ndim);
```

Comment:
This is wrong. Please change semantics of GetShapeLikeArgument to return number of _samples_, not dimensions (or both, perhaps). The return value is not used anywhere except tests.